### PR TITLE
Add logs to generator creation

### PR DESCRIPTION
**Pull request by Walid-Sahnoun**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/modifications/GeneratorCreation.java [hunk 1]
```diff
@@ -201,4 +206,264 @@ private void createGeneratorInBusBreaker(VoltageLevel voltageLevel, GeneratorCre
...
+                    "Voltage level"));
+            voltageReports.add(ModificationUtils.getInstance().buildCreationReport(
+                    generatorCreationInfos.getRegulatingTerminalType() + ":"
+                            + generatorCreationInfos.getRegulatingTerminalId(),
+                    "Equipment"));
+        }
+    }
+
+    private void reportGeneratorConnection(GeneratorCreationInfos generatorCreationInfos, Reporter subReporter) {
+        if ((generatorCreationInfos.getVoltageLevelId() != null && generatorCreationInfos.getBusOrBusbarSectionId() != null)
```
Comment (laurgarn) [inline src/main/java/org/gridsuite/modification/server/modifications/GeneratorCreation.java:344]: From my heretic way of writing, I would replace the first `&&` condition by a
```
if (generatorCreationInfos.getVoltageLevelId() == null) {
  return;
}
if (generatorCreationInfos.getBusOrBusbarSectionId() == null) {
  return;
}
```
and then, after the three possible additions,
```
if (!connectionReports.isEmpty()) {
  ModificationUtils.getInstance().reportModifications(subReporter, connectionReports, "ConnectionCreated", CONNECTION);
}
```
May be factorize `ModificationUtils.getInstance()`

File: src/main/java/org/gridsuite/modification/server/modifications/GeneratorCreation.java [hunk 2]
```diff
@@ -46,8 +57,47 @@ public void check(Network network) throws NetworkModificationException {
...
         }
+
+        // check connectivity
         ModificationUtils.getInstance().controlConnectivity(network, modificationInfos.getVoltageLevelId(),
                 modificationInfos.getBusOrBusbarSectionId(), modificationInfos.getConnectionPosition());
+
+        // check min max reactive limits
+        if (modificationInfos.getMinimumReactivePower() != null && modificationInfos.getMaximumReactivePower() != null) {
+            if (Double.isNaN(modificationInfos.getMinimumReactivePower())) {
+                throw new NetworkModificationException(CREATE_GENERATOR_ERROR, "Generator '" + modificationInfos.getEquipmentId() + "' : minimum reactive power is not set");
```
Comment (laurgarn) [inline src/main/java/org/gridsuite/modification/server/modifications/GeneratorCreation.java:68]: What about defining
```
    private static NetworkModificationException makeGeneratorException(String generatorId, String msgSuffix) {
        return new NetworkModificationException(CREATE_GENERATOR_ERROR, "Generator '" + generatorId + msgSuffix);
}
```
and use it like
```
                throw makeGeneratorException(modificationInfos.getEquipmentId(), "' : minimum reactive power is not set");
```
?

File: src/main/java/org/gridsuite/modification/server/modifications/GeneratorCreation.java (hunk 1 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/modifications/GeneratorCreation.java]: Done

File: src/main/java/org/gridsuite/modification/server/modifications/GeneratorCreation.java (hunk 2 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/modification/server/modifications/GeneratorCreation.java]: Done

**Review by laurgarn — APPROVED** [review]
Code OK

**Pull request merged**